src/app/services/api-data.service.ts

```diff
@@ -40,6 +40,29 @@ export class ApiDataService {
     );
   }
 
+  getAllFilesForProject(project: string) {
+    const url = this.hostSetting.host + 'file/_search/?size=100000&q=secondaryProject:' + project;
+    const headers = new HttpHeaders().set('Content-Type', 'application/json');
+    return this.http.get(url, {headers: headers}).pipe(
+      map((data: any) => {
+        return data.hits.hits.map( entry => ({
+          name: entry['_source']['name'],
+          fileId: entry['_id'],
+          experiment: entry['_source']['experiment']['accession'],
+          assayType: entry['_source']['experiment']['assayType'],
+          experimentTarget: entry['_source']['experiment']['target'],
+          run: entry['_source']['run']['accession'],
+          readableSize: entry['_source']['readableSize'],
+          checksum: entry['_source']['checksum'],
+          checksumMethod: entry['_source']['checksumMethod']
+          })
+        );
+      }),
+      retry(3),
+      catchError(this.handleError),
+    );
+  }
+
```

Comment:
Overall this method is fine, just to be more consistent with other methods can you:
1. remove headers (it's not necessary here, as Angular returns json by default)
2. provide interface for returning data
map((data: any) => {
        return data.hits.hits.map( entry => ({
          name: entry['_source']['name'],
          fileId: entry['_id'],
          experiment: entry['_source']['experiment']['accession'],
          assayType: entry['_source']['experiment']['assayType'],
          experimentTarget: entry['_source']['experiment']['target'],
          run: entry['_source']['run']['accession'],
          readableSize: entry['_source']['readableSize'],
          checksum: entry['_source']['checksum'],
          checksumMethod: entry['_source']['checksumMethod']
          } as FileForProjectsTable )

export interface FileForProjectsTable {
  name: string;
  fileId: string;
  experiment: string;
  assayType: string;
  experimentTarget: string;
  run: string;
  readableSize: string;
  checksum: string;
  checksumMethod: string;
}

Reply:
done in c257a2a